app/controllers/orders_controller.rb

```diff
@@ -324,6 +314,32 @@ def index
 
   private
 
+  def add_account_to_order(account)
+    if account
+      success = true
+      @order.transaction do
+        begin
+          @order.invalidate
+          @order.update!(account_id: account.id)
+        rescue ActiveRecord::ActiveRecordError => e
+          success = false
+          raise ActiveRecord::Rollback
```

Comment:
@jossim I'm wondering if hitting this rollback is causing the issue with specs - put a pry in here and see if you have access to the nonbillable account